cardano-tx-generator/src/Cardano/Benchmarking/GeneratorTx/Genesis.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NoMonomorphismRestriction #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# OPTIONS_GHC -Wno-all-missed-specialisations #-}
+
+module Cardano.Benchmarking.GeneratorTx.Genesis
+  ( extractGenesisFunds
+  , genesisExpenditure
+  , keyAddress
+  )
+where
+
+import           Prelude (error)
+import           Cardano.Prelude hiding (TypeError)
+
+import           Control.Arrow ((***))
+import qualified Data.Map.Strict as Map
+
+-- Era-agnostic imports
+import qualified Ouroboros.Consensus.Cardano as Consensus
+
+-- Byron-specific imports
+import qualified Cardano.Chain.Common as Byron
+import qualified Cardano.Chain.UTxO   as Byron
+
+-- Shelley-specific imports
+import qualified Ouroboros.Consensus.Shelley.Ledger.Ledger as Shelley
+
+import Cardano.Api.Typed
+import Cardano.Benchmarking.GeneratorTx.Era
+import Cardano.Benchmarking.GeneratorTx.Tx
+import Cardano.Benchmarking.GeneratorTx.Tx.Byron
+
+
+keyAddress :: Era era -> SigningKeyOf era -> Address era
```

Comment:
would it make sense to define a class with instances for Byron, Shelley that provide these functions?

Reply:
The GADT dispatch is based on the `cardano-api` model, which isn't class-based.